pkg/machine/convert.go

```diff
@@ -375,6 +377,24 @@ func GetAPIV2NodeCloudSpec(machineSpec clusterv1alpha1.MachineSpec) (*apiv1.Node
 	return cloudSpec, nil
 }
 
+func extractKubeVirtOsImageURLOrDataVolumeName(osImage string) string {
```

Comment:
we already have the `extractKubeVirtOsImageURLOrDataVolumeNsName` function in `common`, cannot we merge those two? They works almost the same. You could return more values from the function if needed.

Reply:
I see you cannot return more values, but you can merge functionality of those functions anyway.

Reply:
As agreed, we will remove function `extractKubeVirtOsImageURLOrDataVolumeName` and we will allow to inject the cluster namespace afterwards. It's better option than introducing part of the code that will have to be removed in the next release due to our plans.